.github/workflows/check-changelog.yml

```diff
@@ -0,0 +1,42 @@
+name: Check Changelog
+# This check makes sure that the changelog is properly updated
+# when a PR introduces a change in a test file.
+# To bypass this check, label the PR with "No Changelog Needed".
+on:
+  pull_request:
+
+jobs:
+  check:
+    runs-on: ubuntu-latest
+    if: ${{ contains(github.event.pull_request.labels.*.name, 'No Changelog Needed') == 0 }}
+    steps:
+      - name: Get PR number and milestone
+        run: |
+          echo "PR_NUMBER=${{ github.event.pull_request.number }}" >> $GITHUB_ENV
+          echo "MILESTONE_TITLE=${{ github.event.pull_request.milestone.title }} >> $GITHUB_ENV
+      - uses: actions/checkout@v2
+        with:
+          fetch-depth: '0'
+      - name: Check the changelog only if tests have been modified
+        run: |
+          set -xe
+          tests=$(git diff --name-only origin/master)
+          if [[ "$tests" =~ tests ]]
```

Comment:
maybe exit early to avoid too many nesting of if blocks

Reply:
I'm sorry I didn't understand this comment... :(